### PR TITLE
Dataset ids

### PR DESCRIPTION
**Pull request by mbaudis**
This adds ```name```, ```guid``` etc. to the ```Dataset``` object. Still just a skeleton implementation.

### CODE REVIEW

File: src/main/resources/avro/metadata.avdl [hunk 1]
```diff
@@ -122,10 +122,25 @@ For data accessors, they're a simple way to scope exploration and analysis
 what's the distribution of that result in the data from our project?)   
 */
 record Dataset {
-  /** The dataset UUID. This is globally unique. */
+  /**
+  The dataset's id. This is the local/legacy/generic id.
```
Comment (jeromekelleher) [inline src/main/resources/avro/metadata.avdl]: I would leave out the second sentence in this comment. We will (hopefully) have a high-level document explaining the distinction between an ``id`` (a server generated string that is unique and persistent within a particular implementation, but not portable across implementations), and a ``name`` (a user specified string which is intended to provide a more human friendly handle for objects).

File: src/main/resources/avro/metadata.avdl [hunk 2]
```diff
@@ -122,10 +122,25 @@ For data accessors, they're a simple way to scope exploration and analysis
...
+  */
   string id;
 
-  /** A description of the dataset. */
+  /**
+  The dataset's guid. This is globally unique id (UUIDv4).
+  TODO: Required?!
+  */
```
Comment (jeromekelleher) [inline src/main/resources/avro/metadata.avdl]: I think an overall strategy for assigning guids to objects should be devised, which looks at _all_ the types before we add it here. I would favour removing this for now, and explicitly tackling GUIDs across the board. I don't really see any advantage to the data set having a GUID if ReadGroupSets and VariantSets don't.

**Comment by jeromekelleher** [comment]
Looks great, thanks @mbaudis! I've made a few comments above --- basically, I think we need to clarify id/name/guid at the top level. @diekhans, I'm sure you have something to say on this...

**Comment by mbaudis** [comment]
@jeromekelleher 

     basically, I think we need to clarify id/name/guid at the top level.

Exactly; see my comment on https://github.com/ga4gh/schemas/pull/397 and elsewhere ...

**Comment by mbaudis** [comment]
@jeromekelleher Also, keep in mind that we're working on the metadata branch https://github.com/ga4gh/schemas/blob/metadata/src/main/resources/avro/metadata.avdl

File: src/main/resources/avro/metadata.avdl (hunk 2 above)
Comment (dglazer) [inline src/main/resources/avro/metadata.avdl]: I agree -- I like the idea of an API-wide GUID strategy; let's not go piecemeal until we have one.

Comment (diekhans) [inline src/main/resources/avro/metadata.avdl]: agreed @dglazer, a GUID system needs to be consistent across all
GA4GH object in order to implement a provenance system.

Comment (mbaudis) [inline src/main/resources/avro/metadata.avdl]: @dglazer @diekhans Please see https://github.com/ga4gh/schemas/pull/397 (and the metadata branch).

Comment (dglazer) [inline src/main/resources/avro/metadata.avdl]: I've seen it, and I think there's good discussions/work going on there, that's not yet ready to merge into master. I think you'll quickly get consensus on cleaning up the `id` comment and adding a `name` field now. Then the full-on guid conversation can continue at its own pace.

Comment (diekhans) [inline src/main/resources/avro/metadata.avdl]: @mbaudis, we need to have some API-wide design work done before
we defining any new features like GUIDs.  GA4GH doesn't even
document the meaning of `id' and `name', which has lead to
a lot of confusion.

For now, pull request should be the absolute minimum required to
get the baseline functionality working.

**Comment by mbaudis** [comment]
@jeromekelleher @diekhans @dglazer @helenp
I've changed the PR to just add ```name``` to ```Dataset``` (and change the ```id`` comment to *locally*. Pls. vote, for moving this into master.

**Comment by helenp** [comment]
+1

File: src/main/resources/avro/metadata.avdl [hunk 3]
```diff
@@ -9,7 +9,7 @@ protocol Metadata {
 import idl "common.avdl";
 
 /**
-An experimental preparation of a `Sample`.
+An experimental preparation of a `Biosample`.
 */
```
Comment (jeromekelleher) [inline src/main/resources/avro/metadata.avdl]: Are these Biosample comments intentional? We don't have this defined, so perhaps it's better to revert these to Sample (even though that's not defined either!).

**Comment by dglazer** [comment]
I agree with @jeromekelleher re reverting the "biosample" comment changes.

Other than that, +1.

**Comment by mbaudis** [comment]
@dglazer @jeromekelleher Reverted the comments, but not feeling that good about something logically wrong & just doing it for procedural reasons.

```Sample``` will be ```Biosample``` (is so already on the metadata branch, with PR and votes); and the comment reference ```Sample``` from the ```Experiment``` object is void (as of right now) which wouldn't be different when changing it to any other name. Just going with this would have saved some overhead, even if I didn't document the reasoning.

**Pull request merged**